### PR TITLE
Drag bug fix in Scatter and Label

### PR DESCRIPTION
**Pull request by ssunkara1**
Fixes issue #338 

Thanks Daniel for pointing this out.

### CODE REVIEW

**Comment by dmadeka** [comment]
Can you clean up the commit history?

**Comment by ssunkara1** [comment]
@dmadeka Yeah was in the process of doing that. Now it is done.

File: js/src/Scatter.js [hunk 1]
```diff
@@ -993,6 +993,8 @@ var Scatter = mark.Mark.extend({
             // saving on move if flag is set
             this.update_array(d, i);
         }
+        var i = 0;
+        i = i + 1;
```
Comment (dmadeka) [inline js/src/Scatter.js]: Let's make it one commit so this isn't there

Comment (dmadeka) [inline js/src/Scatter.js]: Because you delete it in the second commit

Comment (ssunkara1) [inline js/src/Scatter.js]: @dmadeka Merged the two commits.

**Pull request merged**